### PR TITLE
Add auto-sizes generator tag.

### PR DESCRIPTION
**Pull request by felixarntz**
## Summary

This PR adds a generator meta tag to the remaining plugin that didn't have one yet, `auto-sizes`. Per #1103, it uses the immutable plugin slug for it.

This will be included in the 1.0.1 version (for which the codebase has already been bumped before).

### CODE REVIEW

**Review by thelovekesh — APPROVED** [review]

File: tests/plugins/auto-sizes/auto-sizes-test.php [hunk 1]
```diff
@@ -87,4 +87,16 @@ public function test_content_image_without_lazy_loading_does_not_have_auto_sizes
 			wp_filter_content_tags( $this->get_image_tag( self::$image_id ) )
 		);
 	}
+
+	/**
+	 * Test printing the meta generator tag.
+	 *
+	 * @covers ::auto_sizes_render_generator
+	 */
+	public function test_auto_sizes_render_generator() {
```
Comment (joemcgill) [inline tests/plugins/auto-sizes/auto-sizes-test.php:102]: I'd be for adding an additional test to ensure this callback is properly hooked to the `wp_head` action, similar to [this example](https://github.com/WordPress/performance/pull/1103#discussion_r1546577164).

**Review by joemcgill — APPROVED** [review]
Looks good to me. Pre-approving with one suggestion.

File: tests/plugins/auto-sizes/auto-sizes-test.php (hunk 1 above)
Comment (felixarntz) [reply tests/plugins/auto-sizes/auto-sizes-test.php:102]: Fair point, we can add this here. Though then I'd say lets also assert the other hooks in the same test.

Comment (felixarntz) [reply tests/plugins/auto-sizes/auto-sizes-test.php:102]: Added in https://github.com/WordPress/performance/pull/1105/commits/1a7016145126dd971004d705f0343984d3fae990

**Comment by felixarntz** [comment]
There's an unrelated "Spell Check" CI failure here now, for which I opened #1106. Please take a look.

**Review by joemcgill — APPROVED** [review]
Thanks, @felixarntz!

**Comment by mukeshpanchal27** [comment]
> There's an unrelated "Spell Check" CI failure here now, for which I opened #1106. Please take a look.

The PR has been merged now, and the CI failure has also been fixed in this branch. It's ready for merge.

**Review by mukeshpanchal27 — APPROVED** [review]

**Pull request merged**